packages/in_app_purchase/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -66,6 +69,39 @@ void setActivity(@Nullable Activity activity) {
     this.activity = activity;
   }
 
+  // Activity Life Cycle callbacks
+  @Override
+  public void onActivityCreated(Activity activity, Bundle savedInstanceState) {}
+
+  @Override
+  public void onActivityStarted(Activity activity) {}
+
+  @Override
+  public void onActivityResumed(Activity activity) {}
+
+  @Override
+  public void onActivityPaused(Activity activity) {}
+
+  @Override
+  public void onActivitySaveInstanceState(Activity activity, Bundle outState) {}
+
+  @Override
+  public void onActivityDestroyed(Activity activity) {
+    if (this.activity == activity) {
+      if (this.applicationContext != null) {
```

Comment:
style nit: I would combine this with the outer `if` just to avoid an extra level of indentation.

Reply:
done